File: Two Pointers/longest-palindromic-substring/submission-0.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        
        maxLen = 0
        result = ""
        for i in range(len(s)):
            # Odd
            l,r = i,i
            while l > -1 and r < len(s) and s[l] == s[r]:
                if len(s[l:r+1]) > maxLen:
                    result = s[l:r+1]
                    maxLen = len(s[l:r+1])
                l = l - 1
                r = r + 1

            # Even
            l,r = i, i+1
            while  l > -1 and r < len(s) and s[l] == s[r]:
                if len(s[l:r+1]) > maxLen:
                    result = s[l:r+1]
                    maxLen = len(s[l:r+1])
                l = l - 1
                r = r + 1



        return result

File: Math/climbing-stairs/submission-0.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n <= 3) {
            return n;
        }

        auto prev_1 = 2;
        auto prev_2 = 1;
        auto result = 0;
        for (auto i = 3; i <= n; i++) {
            result = prev_1 + prev_2;
            prev_2 = prev_1;
            prev_1 = result;

        }
        
        return result;
    }
};